packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -394,6 +401,53 @@ ListButton.propTypes = {
   }).isRequired,
 };
 
+const LinkButton = () => {
+  const editor = useSlate();
+
+  const isLinkActive = () => {
+    if (!editor.selection) return false;
+
+    const [match] = Array.from(
+      Editor.nodes(editor, {
+        at: Editor.unhangRange(editor, editor.selection),
+        match: (node) =>
+          !Editor.isEditor(node) && SlateElement.isElement(node) && node.type === 'link',
+      })
+    );
+
+    return Boolean(match);
+  };
+
+  const isActive = isLinkActive();
+
+  const addLink = () => {
+    // We insert an empty anchor, so we split the DOM to have a element we can use as reference for the popover
+    insertLink(editor, { url: '', text: Editor.string(editor, editor.selection) });
+  };
+
+  const selectionIsCollapsed = editor.selection && Range.isCollapsed(editor.selection);
+  const selectionIsInSameBlock =
+    editor.selection &&
+    Path.equals(
+      Path.parent(editor.selection.focus.path),
+      Path.parent(editor.selection.anchor.path)
+    );
+
+  return (
+    <ToolbarButton
+      icon={Link}
+      name="link"
+      label={{
+        id: 'components.Blocks.link',
+        defaultMessage: 'Link',
+      }}
+      isActive={isActive}
+      handleClick={addLink}
+      disabled={selectionIsCollapsed || !selectionIsInSameBlock}
```

Comment:
not sure what is this 2nd condition for, but selectionIsCollapsed also checking the same and 1 step more, but you are negating 2nd condition, its confusing